src/diffusers/pipelines/latent_diffusion/pipeline_latent_diffusion_superresolution.py

```diff
@@ -0,0 +1,165 @@
+import inspect
+from typing import Optional, Tuple, Union
+
+import numpy as np
+import torch
+import torch.utils.checkpoint
+
+import PIL
+
+from ...models import UNet2DModel, VQModel
+from ...pipeline_utils import DiffusionPipeline, ImagePipelineOutput
+from ...schedulers import (
+    DDIMScheduler,
+    EulerAncestralDiscreteScheduler,
+    EulerDiscreteScheduler,
+    LMSDiscreteScheduler,
+    PNDMScheduler,
+)
+
+
+def preprocess(image):
+    w, h = image.size
+    w, h = map(lambda x: x - x % 32, (w, h))  # resize to integer multiple of 32
```

Comment:
Why is this necessary? An alternative would be to pad and then crop the upscaled image. Not sure if it's worth it, slightly worried that this might skew images a little bit.

Reply:
@pcuenca This is how other pipelines resize the image so it can successfully forward over UNet. Just copied it. Really sorry I can't fully understand your suggestion, could you kindly push a commit for it?

Reply:
Here the preprocessing should be similar to how it's done in the original repo, since the model is trained on the preprocessed image. @duongna21 could post a link to the original inference code ?

Reply:
@patil-suraj Look at [this](https://github.com/CompVis/stable-diffusion/blob/69ae4b35e0a0f6ee1af8bb9a5d0016ccb27e36dc/notebook_helpers.py#L148-L161) and [this](https://github.com/CompVis/stable-diffusion/blob/ce05de28194041e030ccfc70c635fe3707cdfc30/ldm/models/diffusion/ddpm.py#L902-L984). It works great with varying img size.

Reply:
Thanks and no worries. We'll try to take a look at this, we can merge the PR without that also.